https://bugs.gentoo.org/542638

From: Stephan Wolf <stephan@letzte-bankreihe.de>

Samba uses the build in implementation of heimdal from year 2011.
The ebuild uses the portage version of heimdal.
I got it fixed to patch samba kdc_init function.

--- samba-4.2.3/source4/kdc/kdc.c
+++ samba-4.2.3/source4/kdc/kdc.c
@@ -967,9 +967,9 @@
 	 * The old behavior in the _kdc_get_preferred_key()
 	 * function is use_strongest_server_key=TRUE.
 	 */
-	kdc->config->as_use_strongest_session_key = false;
+	kdc->config->tgt_use_strongest_session_key = false;
 	kdc->config->preauth_use_strongest_session_key = false;
-	kdc->config->tgs_use_strongest_session_key = false;
+	kdc->config->svc_use_strongest_session_key = false;
 	kdc->config->use_strongest_server_key = true;
 
 	/* Register hdb-samba4 hooks for use as a keytab */
